src/v/storage/disk_log_impl.h

```diff
@@ -146,7 +146,8 @@ class disk_log_impl final : public log {
 
     size_t size_bytes() const override { return _probe->partition_size(); }
     uint64_t size_bytes_after_offset(model::offset o) const override;
-    ss::future<> update_configuration(ntp_config::default_overrides) final;
+    void set_overrides(ntp_config::default_overrides) final;
+    bool notify_compaction_update() final;
```

Comment:
I'm not sure why this was split, it seems easier to misuse now.

Reply:
These two manage independent behaviours. When the cluster-level config `log_cleanup_policy` changes there is no `ntp_config::default_overrides` to update but the compaction status still needs to update. I was following the pattern used in the consensus layer (sitting between `partition` and `log`) which has a similar `notify_config_update` method for write caching config changes:
https://github.com/redpanda-data/redpanda/blob/1e1557564a4990de37434e36cb1e6baaeaf5f1d2/src/v/raft/consensus.h#L516-L517
It's only used in `partition.cc` and partition+consensus+log are already highly coupled, so I think this is fine. (Perhaps the ideal way would be to have a methot with a signature `ss::future<change_result> apply_config_change(std::variant<topic_config_change, cluster_config_change>);` but I wanted to keep the refactoring minimal and consistent with the existing pattern.)

Reply:
> (Perhaps the ideal way would be to have a methot with a signature ss::future<change_result> apply_config_change(std::variant<topic_config_change, cluster_config_change>); but I wanted to keep the refactoring minimal and consistent with the existing pattern.)

The concern here is that splitting it up means that the two calls must both be made or there's a bug. Within the log, it's not clear whether to read `_compaction_enabled` or `config().is_compacted()`.

It might be worth changing the name, or making it an optional that's only engaged during the gap between the two calls.